playground/frontend/playground_components/lib/src/controllers/playground_controller.dart

```diff
@@ -137,8 +140,38 @@ class PlaygroundController with ChangeNotifier {
       selectedExample?.type != ExampleType.test &&
       [Sdk.java, Sdk.python].contains(sdk);
 
+  /// If no SDK is selected, sets it to [sdk] and creates an empty state for it.
+  void setEmptyIfNoSdk(Sdk sdk) {
```

Comment:
All users of this method set `defaultSdk` if their own value is null. Suggest to move this logic here

Reply:
This is `playground_components` package, it has no notion of a default SDK.